### PR TITLE
UCS/CONFIG: During exit, issue warning if a UCX_ env var was never used

### PR DESCRIPTION
**Pull request by yosefe**

### CODE REVIEW

File: src/ucs/config/parser.c [hunk 1]
```diff
@@ -1237,20 +1263,9 @@ void ucs_config_parser_check_env_vars()
...
-                found = 1;
-                break;
-            }
-        }
-
-        if (!found) {
-            ucs_warn("invalid configuration variable '%s'", var_name);
+        iter = kh_get(ucs_config_env_vars, &ucs_config_parser_env_vars, var_name);
+        if (iter == kh_end(&ucs_config_parser_env_vars)) {
+            ucs_warn("unused environment variable: '%s'", var_name);
```
Comment (yosefe) [inline src/ucs/config/parser.c]: one line for all vars + info how to suppress

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/6577/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/6581/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/6585/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/6590/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
@alinask pls take a look

**Comment by alinask** [comment]
If a user has UCX_ param set in his env (like UCX_HOME for example), would a warning be printing about it?

**Pull request merged**